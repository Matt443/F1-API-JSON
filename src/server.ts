import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}