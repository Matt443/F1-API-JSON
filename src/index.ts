import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
