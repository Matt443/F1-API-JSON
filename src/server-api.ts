import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
