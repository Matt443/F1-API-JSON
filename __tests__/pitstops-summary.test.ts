import { getPitstopsSummary } from "../src/server";

test("fastest laps", async () => {
    expect(await getPitstopsSummary(2024)).toMatchSnapshot();
});
