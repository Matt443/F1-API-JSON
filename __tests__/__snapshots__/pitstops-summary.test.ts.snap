// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fastest laps 1`] = `
[
  {
    "code": "RIC",
    "lap": 5,
    "name": "Daniel Ricciardo",
    "number": 3,
    "stops": "1",
    "team": "RB Honda RBPT",
    "time": "18.265",
    "timeOfDay": 2024-03-24T15:10:25.000Z,
    "total": "18.265",
  },
  {
    "code": "ALB",
    "lap": 6,
    "name": "Alexander Albon",
    "number": 23,
    "stops": "1",
    "team": "Williams Mercedes",
    "time": "17.755",
    "timeOfDay": 2024-03-24T15:11:45.000Z,
    "total": "17.755",
  },
  {
    "code": "ZHO",
    "lap": 6,
    "name": "Zhou Guanyu",
    "number": 24,
    "stops": "1",
    "team": "Kick Sauber Ferrari",
    "time": "19.032",
    "timeOfDay": 2024-03-24T15:11:49.000Z,
    "total": "19.032",
  },
  {
    "code": "HAM",
    "lap": 7,
    "name": "Lewis Hamilton",
    "number": 44,
    "stops": "1",
    "team": "Mercedes",
    "time": "17.596",
    "timeOfDay": 2024-03-24T15:13:04.000Z,
    "total": "17.596",
  },
  {
    "code": "MAG",
    "lap": 7,
    "name": "Kevin Magnussen",
    "number": 20,
    "stops": "1",
    "team": "Haas Ferrari",
    "time": "18.301",
    "timeOfDay": 2024-03-24T15:13:08.000Z,
    "total": "18.301",
  },
  {
    "code": "RUS",
    "lap": 8,
    "name": "George Russell",
    "number": 63,
    "stops": "1",
    "team": "Mercedes",
    "time": "18.046",
    "timeOfDay": 2024-03-24T15:14:23.000Z,
    "total": "18.046",
  },
  {
    "code": "STR",
    "lap": 8,
    "name": "Lance Stroll",
    "number": 18,
    "stops": "1",
    "team": "Aston Martin Aramco Mercedes",
    "time": "17.901",
    "timeOfDay": 2024-03-24T15:14:26.000Z,
    "total": "17.901",
  },
  {
    "code": "BOT",
    "lap": 8,
    "name": "Valtteri Bottas",
    "number": 77,
    "stops": "1",
    "team": "Kick Sauber Ferrari",
    "time": "46.969",
    "timeOfDay": 2024-03-24T15:14:30.000Z,
    "total": "46.969",
  },
  {
    "code": "LEC",
    "lap": 9,
    "name": "Charles Leclerc",
    "number": 16,
    "stops": "1",
    "team": "Ferrari",
    "time": "17.409",
    "timeOfDay": 2024-03-24T15:15:42.000Z,
    "total": "17.409",
  },
  {
    "code": "PIA",
    "lap": 9,
    "name": "Oscar Piastri",
    "number": 81,
    "stops": "1",
    "team": "McLaren Mercedes",
    "time": "17.573",
    "timeOfDay": 2024-03-24T15:15:43.000Z,
    "total": "17.573",
  },
  {
    "code": "TSU",
    "lap": 9,
    "name": "Yuki Tsunoda",
    "number": 22,
    "stops": "1",
    "team": "RB Honda RBPT",
    "time": "18.523",
    "timeOfDay": 2024-03-24T15:15:50.000Z,
    "total": "18.523",
  },
  {
    "code": "OCO",
    "lap": 9,
    "name": "Esteban Ocon",
    "number": 31,
    "stops": "1",
    "team": "Alpine Renault",
    "time": "18.287",
    "timeOfDay": 2024-03-24T15:15:55.000Z,
    "total": "18.287",
  },
  {
    "code": "NOR",
    "lap": 14,
    "name": "Lando Norris",
    "number": 4,
    "stops": "1",
    "team": "McLaren Mercedes",
    "time": "17.776",
    "timeOfDay": 2024-03-24T15:22:37.000Z,
    "total": "17.776",
  },
  {
    "code": "PER",
    "lap": 14,
    "name": "Sergio Perez",
    "number": 11,
    "stops": "1",
    "team": "Red Bull Racing Honda RBPT",
    "time": "18.112",
    "timeOfDay": 2024-03-24T15:22:45.000Z,
    "total": "18.112",
  },
  {
    "code": "SAI",
    "lap": 16,
    "name": "Carlos Sainz",
    "number": 55,
    "stops": "1",
    "team": "Ferrari",
    "time": "17.599",
    "timeOfDay": 2024-03-24T15:25:14.000Z,
    "total": "17.599",
  },
  {
    "code": "OCO",
    "lap": 16,
    "name": "Esteban Ocon",
    "number": 31,
    "stops": "2",
    "team": "Alpine Renault",
    "time": "30.626",
    "timeOfDay": 2024-03-24T15:25:58.000Z,
    "total": "48.913",
  },
  {
    "code": "ALO",
    "lap": 17,
    "name": "Fernando Alonso",
    "number": 14,
    "stops": "1",
    "team": "Aston Martin Aramco Mercedes",
    "time": "18.160",
    "timeOfDay": 2024-03-24T15:27:06.000Z,
    "total": "18.160",
  },
  {
    "code": "HUL",
    "lap": 17,
    "name": "Nico Hulkenberg",
    "number": 27,
    "stops": "1",
    "team": "Haas Ferrari",
    "time": "18.320",
    "timeOfDay": 2024-03-24T15:27:28.000Z,
    "total": "18.320",
  },
  {
    "code": "GAS",
    "lap": 17,
    "name": "Pierre Gasly",
    "number": 10,
    "stops": "1",
    "team": "Alpine Renault",
    "time": "20.316",
    "timeOfDay": 2024-03-24T15:27:32.000Z,
    "total": "20.316",
  },
  {
    "code": "ALB",
    "lap": 27,
    "name": "Alexander Albon",
    "number": 23,
    "stops": "2",
    "team": "Williams Mercedes",
    "time": "17.745",
    "timeOfDay": 2024-03-24T15:41:35.000Z,
    "total": "35.500",
  },
  {
    "code": "RIC",
    "lap": 29,
    "name": "Daniel Ricciardo",
    "number": 3,
    "stops": "2",
    "team": "RB Honda RBPT",
    "time": "17.751",
    "timeOfDay": 2024-03-24T15:44:29.000Z,
    "total": "36.016",
  },
  {
    "code": "MAG",
    "lap": 33,
    "name": "Kevin Magnussen",
    "number": 20,
    "stops": "2",
    "team": "Haas Ferrari",
    "time": "18.151",
    "timeOfDay": 2024-03-24T15:49:58.000Z,
    "total": "36.452",
  },
  {
    "code": "LEC",
    "lap": 34,
    "name": "Charles Leclerc",
    "number": 16,
    "stops": "2",
    "team": "Ferrari",
    "time": "17.308",
    "timeOfDay": 2024-03-24T15:50:35.000Z,
    "total": "34.717",
  },
  {
    "code": "PER",
    "lap": 35,
    "name": "Sergio Perez",
    "number": 11,
    "stops": "2",
    "team": "Red Bull Racing Honda RBPT",
    "time": "17.222",
    "timeOfDay": 2024-03-24T15:52:18.000Z,
    "total": "35.334",
  },
  {
    "code": "HUL",
    "lap": 35,
    "name": "Nico Hulkenberg",
    "number": 27,
    "stops": "2",
    "team": "Haas Ferrari",
    "time": "18.145",
    "timeOfDay": 2024-03-24T15:52:40.000Z,
    "total": "36.465",
  },
  {
    "code": "ZHO",
    "lap": 35,
    "name": "Zhou Guanyu",
    "number": 24,
    "stops": "2",
    "team": "Kick Sauber Ferrari",
    "time": "36.866",
    "timeOfDay": 2024-03-24T15:52:50.000Z,
    "total": "55.898",
  },
  {
    "code": "TSU",
    "lap": 36,
    "name": "Yuki Tsunoda",
    "number": 22,
    "stops": "2",
    "team": "RB Honda RBPT",
    "time": "17.535",
    "timeOfDay": 2024-03-24T15:53:59.000Z,
    "total": "36.058",
  },
  {
    "code": "BOT",
    "lap": 36,
    "name": "Valtteri Bottas",
    "number": 77,
    "stops": "2",
    "team": "Kick Sauber Ferrari",
    "time": "18.938",
    "timeOfDay": 2024-03-24T15:54:30.000Z,
    "total": "1:05.907",
  },
  {
    "code": "STR",
    "lap": 37,
    "name": "Lance Stroll",
    "number": 18,
    "stops": "2",
    "team": "Aston Martin Aramco Mercedes",
    "time": "17.977",
    "timeOfDay": 2024-03-24T15:55:19.000Z,
    "total": "35.878",
  },
  {
    "code": "PIA",
    "lap": 39,
    "name": "Oscar Piastri",
    "number": 81,
    "stops": "2",
    "team": "McLaren Mercedes",
    "time": "18.533",
    "timeOfDay": 2024-03-24T15:57:32.000Z,
    "total": "36.106",
  },
  {
    "code": "NOR",
    "lap": 40,
    "name": "Lando Norris",
    "number": 4,
    "stops": "2",
    "team": "McLaren Mercedes",
    "time": "17.653",
    "timeOfDay": 2024-03-24T15:58:46.000Z,
    "total": "35.429",
  },
  {
    "code": "SAI",
    "lap": 41,
    "name": "Carlos Sainz",
    "number": 55,
    "stops": "2",
    "team": "Ferrari",
    "time": "17.740",
    "timeOfDay": 2024-03-24T15:59:57.000Z,
    "total": "35.339",
  },
  {
    "code": "ALO",
    "lap": 41,
    "name": "Fernando Alonso",
    "number": 14,
    "stops": "2",
    "team": "Aston Martin Aramco Mercedes",
    "time": "17.995",
    "timeOfDay": 2024-03-24T16:00:37.000Z,
    "total": "36.155",
  },
  {
    "code": "GAS",
    "lap": 41,
    "name": "Pierre Gasly",
    "number": 10,
    "stops": "2",
    "team": "Alpine Renault",
    "time": "17.682",
    "timeOfDay": 2024-03-24T16:01:21.000Z,
    "total": "37.998",
  },
  {
    "code": "OCO",
    "lap": 42,
    "name": "Esteban Ocon",
    "number": 31,
    "stops": "3",
    "team": "Alpine Renault",
    "time": "18.007",
    "timeOfDay": 2024-03-24T16:02:51.000Z,
    "total": "1:06.920",
  },
  {
    "code": "RUS",
    "lap": 45,
    "name": "George Russell",
    "number": 63,
    "stops": "2",
    "team": "Mercedes",
    "time": "17.964",
    "timeOfDay": 2024-03-24T16:06:07.000Z,
    "total": "36.010",
  },
]
`;
